### PR TITLE
Improve & fix widget sizing code

### PR DESCRIPTION
**Pull request by maniac103**
After #650, #445 resurfaced; this PR should fix it.

Also remove some weird manual fiddling with widget sizing.

### CODE REVIEW

**Comment by maniac103** [comment]
@mueller-ma @lolodomo Can you please give this a try?

File: mobile/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -10,4 +10,5 @@
     <dimen name="widgetlist_item_left_margin">0dp</dimen>
     <dimen name="widgetlist_item_right_margin">0dp</dimen>
     <dimen name="widget_margin">0dp</dimen>
+    <dimen name="webview_row_height">36dp</dimen>
```
Comment (maniac103) [inline mobile/src/main/res/values/dimens.xml]: I'm open for suggestions regarding this value. BasicUI uses 36 pixels per row, which is what I based this on. Old code used 80 pixels per row, which amounts to either 80dp (mdpi), ~53dp (hdpi), 40dp (xhdpi) and ~27dp (xxhdpi). Most devices should be xxhdpi nowadays.

**Comment by mueller-ma** [comment]
I tested it with the multimedia page in demo mode.
Before we had in landscape

1. Radio/Volume
2. Picture with multiple smaller pictures
3. A video which was just a black box on my device
4. Heise via Webview, took up the whole reserved space
5. A pic with "320x240 openHAB", with width equals subpage width (is this a placeholder for anything?). Reintroduced #497

Now we have
1. Radio/Volume
2. Picture with multiple smaller pictures
3. Video is missing
4. Heise via Webview, is only on the half of space of the reserved space
5. A pic with "320x240 openHAB", fixed #497

For the video I think it looks better, but the webview bug should fill up all the reserved space.

**Comment by maniac103** [comment]
The WebView has a fixed height set in the sitemap though. Compare e.g. to its rendering in BasicUI: http://demo.openhab.org:8080/basicui/app?w=0203&sitemap=demo ... it didn't take the full height before either, but a fixed amount of pixels. For the demo sitemap one, height was set to 640px; now it's set to 228dp, which e.g. on xxhdpi amounts to 864px. BasicUI renders it at 228px.

In general: previously, with decreasing display density the WebView was rendered larger. Now it's rendered equally large for all devices, which - if you e.g. have an mdpi device -means it might be rendered smaller for you.
@kaikreuzer [The docs](https://docs.openhab.org/configuration/sitemaps.html#element-type-webview) say the height parameter is supposed to be `the number of element rows to fill`, but AFAICT there's no specification how large one row is supposed to be ... is that correct and the interpretation is up to the client?

I'd be interested in seeing the sitemap + item definition for items 3 and 5 as well; I guess something got broken in there.

**Comment by mueller-ma** [comment]
Before
screenshot_2018-02-02-15-52-52
After
screenshot_2018-02-02-15-54-06

**Comment by maniac103** [comment]
7 inch tablet, I assume? If yes, that's mdpi, thus 640px before and 228px afterwards. Compare to BasicUI on the same phone ... or to a somewhat recent phone :-)

**Comment by mueller-ma** [comment]
I dont really care about the height. You see the space after the webview? There should be the 320*240 pic directly after the webview and not free space.

**Comment by maniac103** [comment]
Ah, now I understand. Will check it.

**Comment by mueller-ma** [comment]
LGTM! Can you resolve the merge conflict?

**Comment by maniac103** [comment]
Fixed.

**Pull request merged**